bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerMetadata.java

```diff
@@ -72,14 +73,13 @@
     public static final String VERSION_KEY = "BookieMetadataFormatVersion";
 
     private int metadataFormatVersion = 0;
-
     private int ensembleSize;
     private int writeQuorumSize;
     private int ackQuorumSize;
     private long length;
     private long lastEntryId;
-    private long ctime;
-    boolean storeSystemtimeAsLedgerCreationTime = false;
+    final Optional<Long> ctime;
+    final Optional<Long> metadataStoreCtime;
```

Comment:
this doesn't sound a right refactor to me. it is also very confusing and hard to track, when you are mapping a write behavior (whether to store system time as ledger cation time) to a read behavior (whether metadata store ctime is set). 

If we are doing a refactor, I would suggest : 

- keep only one ctime in the metadata
- when building ledger metadata, passing it the `storeSystemtimeAsLedgerCreationTime` when building proto ledger metadata. simply because the flag is controlling write time only.

Reply:
This whole thing is a hacky abomination. I made the change because passing storeSystemtimeAsLedgerCreationTime is passing client configuration into metadata object, where the scope of the lifetime of these objects is completely different. Still looks hacky though.

I think the correct way to do this is to pass a flag to the serialize() type method. This is where this storage decision should be taken, just as we take a decision on which version we want to store (which will be part of the binary metadata change).

For now, I'll revert this part of the change, and will deal with it in a later PR.

Reply:
works for me